app/models/v2/report.rb

```diff
@@ -44,12 +86,22 @@ def percent_compliant
     has_many :report_systems, class_name: 'V2::ReportSystem', dependent: nil # rubocop:disable Rails/InverseOf
     has_many :systems, class_name: 'V2::System', through: :report_systems
     has_many :reported_systems, class_name: 'V2::System', through: :test_results, source: :system, dependent: nil
+    # rubocop:disable Rails/InverseOf
+    has_many :reporting_and_non_reporting_systems, lambda {
+      joins(
+        "INNER JOIN inventory.hosts ON inventory.hosts.id = report_systems.system_id \
+          LEFT OUTER JOIN v2_test_results ON v2_test_results.system_id = report_systems.system_id"
+      ).where(
+        Pundit.policy_scope(User.current, V2::System).where_clause.ast.to_sql
```

Comment:
Is this really necessary? Aren't you already doing system scoping on the aggregation level?

Reply:
It does seem like. Investigating...

Reply:
I would need a bit more details here ... because you are scoping down systems to users but then in the aggregation there is additional scoping happening.